bindings_node/src/mls_client.rs

```diff
@@ -0,0 +1,210 @@
+  }
+
+  #[napi]
+  pub fn is_registered(&self) -> bool {
+    self.inner_client.identity().signature_request().is_none()
+  }
+
+  #[napi]
+  pub fn installation_id(&self) -> String {
+    ed25519_public_key_to_address(self.inner_client.installation_public_key().as_slice())
+  }
+
+  #[napi]
+  pub async fn can_message(&self, account_addresses: Vec<String>) -> Result<HashMap<String, bool>> {
+    let results: HashMap<String, bool> = self
+      .inner_client
+      .can_message(account_addresses)
+      .await
+      .map_err(|e| Error::from_reason(format!("{}", e)))?;
+
+    Ok(results)
+  }
+
+  #[napi]
+  pub fn add_ecdsa_signature(&mut self, signature_bytes: Uint8Array) -> Result<()> {
+    if self.is_registered() {
+      return Err(Error::from_reason(
+        "An identity is already registered with this client",
+      ));
+    }
+
+    let signature_text = match self.signature_text() {
+      Some(text) => text,
+      None => return Err(Error::from_reason("No signature text found")),
+    };
+
+    let signature = Box::new(RecoverableEcdsaSignature::new(
+      signature_text,
+      signature_bytes.deref().to_vec(),
+    ));
+
+    self.signatures.insert(
+      MemberIdentifier::Address(self.account_address.clone()),
+      signature,
+    );
+
+    Ok(())
+  }
+
+  #[napi]
+  pub fn add_erc1271_signature(
+    &mut self,
+    signature_bytes: Uint8Array,
+    chain_id: String,
+    account_address: String,
+    chain_rpc_url: String,
+    block_number: BigInt,
+  ) -> Result<()> {
+    if self.is_registered() {
+      return Err(Error::from_reason(
+        "An identity is already registered with this client",
+      ));
+    }
+
+    let signature_text = match self.signature_text() {
+      Some(text) => text,
+      None => return Err(Error::from_reason("No signature text found")),
+    };
+
+    let account_id = AccountId::new(chain_id, account_address.clone());
+
+    let signature = Box::new(Erc1271Signature::new(
+      signature_text,
+      signature_bytes.deref().to_vec(),
+      account_id,
+      chain_rpc_url,
+      block_number.get_u64().1,
+    ));
+
+    self.signatures.insert(
+      MemberIdentifier::Address(account_address.clone()),
+      signature,
+    );
+
+    Ok(())
+  }
+
+  #[napi]
+  pub async fn register_identity(&self) -> Result<()> {
+    if self.is_registered() {
+      return Err(Error::from_reason(
+        "An identity is already registered with this client",
+      ));
+    }
+
+    if self.signatures.is_empty() {
+      return Err(Error::from_reason(
+        "No client signatures found, add at least 1 before registering",
+      ));
+    }
+
+    let mut signature_request = match self.inner_client.identity().signature_request() {
+      Some(signature_req) => signature_req,
+      None => return Err(Error::from_reason("No signature request found")),
+    };
+
+    for signature in self.signatures.values() {
+      signature_request
+        .add_signature(signature.clone())
```

Comment:
This could error because not all in the signatures HashMap is needed. Try using `signature_request.missing_signatures() -> Vec<MemberIdentifier>`?

Reply:
updated the logic here: [Update register_identity](https://github.com/xmtp/libxmtp/pull/800/commits/be09af822c82c667b895a4c6dc9607f9624ba1e0)